oci-artificial-intelligence/ai-language/custom-model/custom-model.md

```diff
@@ -0,0 +1,381 @@
+
+    ![](./images/txtc-review.png " ")
+
+## **Task 4:** Create an Endpoint and Analyze through endpoint.
+
+1. Under Model endpoints, click on create model endpoint
+2. Specify the name of the endpoint (e.g., ticket_classification_endpoint1)
+    
+    ![](./images/create-model-endpoint.png " ")
+
+3. Once endpoint is active, Under Resources, click on Analyze link.
+4. Enter text, and click on Analyze to see the result
+
+    ![](./images/analyze.png " ")
+
+## **Task 5:** Create Custom Models and analyzing through endpoint with Python SDK.
+
+For setup, please refer Lab 1
+
+[Proceed to the next section](#next).
+Follow below steps to run Python SDK:
+
+#### 1. Download Python Code.
+
+#### Custom NER Python Code
+
+```Python
+<copy>
+import time
+import oci
+print(f'OCI Client SDK version: {oci.__version__}')
+config = oci.config.from_file()
+
+compartment_id = "ocid1.tenancy.oc1..aaaaaaaaih4krf4od5g2ym7pffbp6feof3rx64522aoxxvv3iuw3tam6fvea"  #TODO Provide your compartmentId here
+
+project_name = None #"custom-NER-project"
+model_name = None #"custom-NER-model"
+endpoint_name = None #"custom_NER_endpoint"
+
+#assuming ashburn endpoint.
+# For other regions, use below end points
+endpoint = "https://language.aiservice-preprod.us-ashburn-1.oci.oraclecloud.com"
+#Ashburn: https://language.aiservice.us-ashburn-1.oci.oraclecloud.com
+#Phoenix: https://language.aiservice.us-phoenix-1.oci.oraclecloud.com
+#Frankfurt: https://language.aiservice.eu-frankfurt-1.oci.oraclecloud.com
+#London: https://language.aiservice.uk-london-1.oci.oraclecloud.com
+#Mumbai: https://language.aiservice.ap-mumbai-1.oci.oraclecloud.com
+print(f"Instantiating AI Services client with end point: {endpoint}")
+
+ai_client = oci.ai_language.AIServiceLanguageClient(config, service_endpoint=endpoint)
+
+#create ailanguageproject
+project_details = oci.ai_language.models.CreateProjectDetails(compartment_id=compartment_id,display_name=project_name)
+print(f"Creating project with details:{project_details}")
+project = ai_client.create_project(project_details)
+print(f"create_project returned: {project.data}")
+
+#wait till project state becomes ACTIVE
+project_id = project.data.id
+project_status = project.data.lifecycle_state
+while (project.data.lifecycle_state == "CREATING"):
+    print('Waiting for project creation to complete...')
+    time.sleep(1*60) #sleep for 1 minute
+    project = ai_client.get_project(project_id)
+
+project = ai_client.get_project(project_id)
+project_status = project.data.lifecycle_state
+print(f"Project status changed from CREATING to {project_status}")
+
+#creating ailanguagemodel
+location_details = oci.ai_language.models.ObjectListDataset(
+    location_type="OBJECT_LIST", 
```

Comment:
remove the  namespace and ask user to provide the namespace
Suggest crating variables at the top without initializing for compartmentID, namespace, etc for e.g. compartmentID = <TODO - provide the compartmentID required here>
and use in the code

Reply:
Done